### ISSUE TITLE
More jest integration

### ISSUE THREAD

**Issue by umidbekkarimov** [thymikee/snapshot-diff#10]
First of all I would like to thank you for your work! 
It allowed me to decreased snapshot file size from 10k lines to 1.3k!

#### Add custom matcher

e.g. https://github.com/americanexpress/jest-image-snapshot#usage
See https://facebook.github.io/jest/docs/expect.html#expectextendmatchers
Usage: 
```js
expect(nextState).toMatchDiffSnapshot(prevState, { contextLines: 2 })
```

#### Add snapshot serializer based on input type

e.g. https://github.com/prettier/prettier/blob/master/tests_config/raw-serializer.js

Current output for object diffs is too noisy, so I had do implement custom serializer in my project, and now output changed  from:

```
 Object {
   \\"bail\\": false,
   \\"context\\": \\".\\",
   \\"devServer\\": Object {
     \\"disableHostCheck\\": true,
-    \\"host\\": \\"localhost\\",
+    \\"host\\": \\"local.host\\",
     \\"hotOnly\\": true,
     \\"inline\\": true,
     \\"noInfo\\": true,
     \\"overlay\\": Object {
       \\"errors\\": true,
       \\"warnings\\": true,
     },
     \\"port\\": 3000,
```

To:

```
 Object {
   "bail": false,
   "context": ".",
   "devServer": Object {
     "disableHostCheck": true,
-    "host": "localhost",
+    "host": "local.host",
     "hotOnly": true,
     "inline": true,
     "noInfo": true,
     "overlay": Object {
       "errors": true,
       "warnings": true,
     },
     "port": 3000,

```

I would love to add PR with this changes.

**Comment by thymikee** [comment]
Hey! Thanks for posting. And I'd love to review the PR :)
cc @ferrannp

**Comment by thymikee** [comment]
As a side note, I forgot about custom matcher and serializer but this seems like a fine way to go. 
Instead I thought about fixing the issue upstream in Jest – here's a proposal https://github.com/facebook/jest/pull/4183.

**Comment by umidbekkarimov** [comment]
@thymikee should I close this and wait for https://github.com/facebook/jest/pull/4183 or proceed to second step?

**Comment by thymikee** [comment]
I'd go with snapshot serializer now, and switch to facebook/jest#4183 when it's merged and new Jest is released. You can mimic the output I guess.

**Comment by thymikee** [comment]
Fixed with #12
